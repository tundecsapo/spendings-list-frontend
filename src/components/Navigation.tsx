import { ReactNode } from "react";

import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import PaymentsIcon from "@mui/icons-material/Payments";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";

export const Navigation = ({ children }: { children: ReactNode }) => {
  return (
    <>
      <AppBar
        position="fixed"
        style={{ background: "white" }}
        sx={{
          boxShadow: "0 3px 2px -2px gray",
        }}
      >
        <Container maxWidth="xl">
          <Toolbar disableGutters>
            <Box
              sx={(theme) => ({
                display: "flex",
                alignItems: "center",
                color: theme.palette.primary.dark,
              })}
            >
              <Box
                sx={{
                  p: 1,
                  backgroundColor: "#f5f0f9",
                  height: "28px",
                  borderRadius: 2,
                }}
              >
                <PaymentsIcon />
              </Box>
              <Typography
                variant="h6"
                noWrap
                sx={{
                  ml: 2,
                }}
              >
                Spendings
              </Typography>
            </Box>
          </Toolbar>
        </Container>
      </AppBar>
      {children}
    </>
  );
};
