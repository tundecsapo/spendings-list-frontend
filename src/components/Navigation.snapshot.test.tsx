import renderer from "react-test-renderer";
import { Navigation } from "./Navigation";

it("Navigation renders correctly", () => {
  const tree = renderer
    .create(
      <Navigation>
        <p>children</p>
      </Navigation>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
