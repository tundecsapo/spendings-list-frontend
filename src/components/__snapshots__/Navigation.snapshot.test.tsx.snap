// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
[
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-pzzqw0-MuiPaper-root-MuiAppBar-root"
    style={
      {
        "background": "white",
      }
    }
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <div
          className="MuiBox-root css-1vsrqil"
        >
          <div
            className="MuiBox-root css-76hfd0"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PaymentsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 14V6c0-1.1-.9-2-2-2H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zm-9-1c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm13-6v11c0 1.1-.9 2-2 2H4v-2h17V7h2z"
              />
            </svg>
          </div>
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-v5xmk6-MuiTypography-root"
          >
            Spendings
          </h6>
        </div>
      </div>
    </div>
  </header>,
  <p>
    children
  </p>,
]
`;
